test/e2e/addon/nodemonitoringagent.go

```diff
@@ -0,0 +1,261 @@
+package addon
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+
+	"slices"
+
+	"github.com/aws/eks-hybrid/test/e2e/errors"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+const (
+	nodeMonitoringAgentNamespace = "kube-system"
+	nodeMonitoringAgentName      = "eks-node-monitoring-agent"
+	lockupDaemonSetName          = "kernel-soft-lockup"
+	lockupContainerNamespace     = "default"
+)
+
+type NodeMonitoringAgentTest struct {
+	Cluster   string
+	addon     *Addon
+	K8S       clientgo.Interface
+	EKSClient *eks.Client
+	K8SConfig *rest.Config
+	Logger    logr.Logger
+}
+
+func (n *NodeMonitoringAgentTest) Run(ctx context.Context) error {
+	n.addon = &Addon{
+		Cluster:   n.Cluster,
+		Namespace: nodeMonitoringAgentNamespace,
+		Name:      nodeMonitoringAgentName,
+	}
+
+	if err := n.Create(ctx); err != nil {
+		return err
+	}
+
+	if err := n.Validate(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n NodeMonitoringAgentTest) Create(ctx context.Context) error {
+	if err := n.addon.CreateAddon(ctx, n.EKSClient, n.K8S, n.Logger); err != nil {
+		return err
+	}
+
+	if err := kubernetes.WaitForDaemonSetReady(ctx, n.Logger, n.K8S, n.addon.Namespace, n.addon.Name); err != nil {
+		return err
+	}
+
+	if err := n.deployKernelError(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n NodeMonitoringAgentTest) deployKernelError(ctx context.Context) error {
+	labels := map[string]string{"app": lockupDaemonSetName}
+	daemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      lockupDaemonSetName,
+			Namespace: lockupContainerNamespace,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  lockupDaemonSetName,
+							Image: "public.ecr.aws/amazonlinux/amazonlinux:2023-minimal",
+							SecurityContext: &v1.SecurityContext{
+								Privileged: ptr.To(true),
+								RunAsUser:  ptr.To(int64(0)),
+							},
+							Command: []string{
+								"/bin/bash",
+								"-c",
+								"echo 'watchdog: BUG: soft lockup - CPU#6 stuck for 23s! [VM Thread:4054]' | tee -a /host/dev/kmsg",
```

Comment:
this is interesting
Why did you decide to do it with a daemon set vs just looping through the nodes and running a command on each?

Reply:
it didn't occur to me that I could run this on the node. I'll do that instead.